src/App.php

```diff
@@ -89,6 +85,31 @@ class App
 	 */
 	private $mode;
 
+	/**
+	 * @var string The App basepath
+	 */
+	private $basepath;
```

Comment:
If you're going for the `camelCase` instead of `under_score`, please rename to `$basePath`.